internal/helmdeployer/install.go

```diff
@@ -514,3 +519,22 @@ func getDryRunConfig(chart *chartv2.Chart, dryRun bool) dryRunConfig {
 
 	return cfg
 }
+
+// isInDownstreamResources returns true and the matching DownstreamResource
+// when a resource with the provided name exists in the provided
+// BundleDeploymentOptions.DownstreamResources slice. If not found, returns
+// false and the zero value for fleet.DownstreamResource.
+func isInDownstreamResources(resourceName, kind string, options fleet.BundleDeploymentOptions) bool {
+	if !experimental.CopyResourcesDownstreamEnabled() {
+		return false
+	}
+
+	for _, dr := range options.DownstreamResources {
+		if dr.Name == resourceName {
+			if dr.Kind == kind {
```

Comment:
nit: this could be a single condition:
```suggestion
		if dr.Name == resourceName && dr.Kind == kind {
```